platform/pulpcore/app/models/content.py

```diff
@@ -79,27 +79,21 @@ class Artifact(Model):
         downloaded (models.BooleanField): The associated file has been successfully downloaded.
         requested (models.BooleanField): The associated file has been requested by a client at
             least once.
-        relative_path (models.TextField): The artifact's path relative to the associated
-            :class:`Content`. This path is incorporated in the absolute storage path of the file
-            and its published path relative to the root publishing directory. At a minimum the path
-            will contain the file name but may also include sub-directories.
         size (models.IntegerField): The size of the file in bytes.
         md5 (models.CharField): The MD5 checksum of the file.
         sha1 (models.CharField): The SHA-1 checksum of the file.
         sha224 (models.CharField): The SHA-224 checksum of the file.
         sha256 (models.CharField): The SHA-256 checksum of the file.
         sha384 (models.CharField): The SHA-384 checksum of the file.
         sha512 (models.CharField): The SHA-512 checksum of the file.
-
-    Relations:
-
-        content (models.ForeignKey): The associated content.
     """
+    def storage_path(self, name):
+        digest = self.sha256
+        return "%s/artifacts/%s/%s" % (settings.MEDIA_ROOT, digest[0:2], digest[2:])
```

Comment:
Good points. I was generally thinking that this type of logic is a good fit for being handled within the domain of the FileField object, and I think that lines up with your specific suggestions.